R/dashboard_modules/04-learner_characteristics.R

```diff
@@ -9,13 +9,80 @@ chars_parquet <- read_chars("data/apprenticeships_demographics_0.parquet") %>%
   ) %>%
   mutate(measure = firstup(measure))
 
+# this pivots it longer still and puts in the totals for the table
+chars_parquet_total_lldd <- chars_parquet %>%
+  filter(lldd == "Total" & sex == "Total" & age_group == "Total" & ethnicity_major == "Total") %>%
+  mutate(
+    characteristic_type = "Learner with learning difficulties or disabilities (LLDD)",
+    characteristic = "Total"
+  ) %>%
+  select(year, provider_name, characteristic_type, characteristic, measure, count)
+
+chars_parquet_lldd <- chars_parquet %>%
+  filter(lldd != "Total") %>%
+  mutate(
+    characteristic_type = "Learner with learning difficulties or disabilities (LLDD)",
+    characteristic = lldd
+  ) %>%
+  select(year, provider_name, characteristic_type, characteristic, measure, count)
 
-age_parquet <- chars_parquet %>%
-  filter(lldd == "Total" & sex == "Total" & ethnicity_major == "Total" & count > 0)
 
-ethnicity_parquet <- chars_parquet %>%
-  filter(lldd == "Total" & sex == "Total" & age_group == "Total" & count > 0)
+chars_parquet_total_sex <- chars_parquet %>%
+  filter(lldd == "Total" & sex == "Total" & age_group == "Total" & ethnicity_major == "Total") %>%
+  mutate(
+    characteristic_type = "Sex",
+    characteristic = sex
+  ) %>%
+  select(year, provider_name, characteristic_type, characteristic, measure, count)
+
+chars_parquet_sex <- chars_parquet %>%
+  filter(sex != "Total") %>%
+  mutate(
+    characteristic_type = "Sex",
+    characteristic = sex
+  ) %>%
+  select(year, provider_name, characteristic_type, characteristic, measure, count)
 
+chars_parquet_total_age <- chars_parquet %>%
+  filter(lldd == "Total" & sex == "Total" & age_group == "Total" & ethnicity_major == "Total") %>%
+  mutate(
+    characteristic_type = "Age",
+    characteristic = age_group
+  ) %>%
+  select(year, provider_name, characteristic_type, characteristic, measure, count)
+
+chars_parquet_age <- chars_parquet %>%
+  filter(age_group != "Total") %>%
+  mutate(
+    characteristic_type = "Age",
+    characteristic = age_group
+  ) %>%
+  select(year, provider_name, characteristic_type, characteristic, measure, count)
+
+chars_parquet_total_ethnicity <- chars_parquet %>%
+  filter(lldd == "Total" & sex == "Total" & age_group == "Total" & ethnicity_major == "Total") %>%
+  mutate(
+    characteristic_type = "Ethnicity",
+    characteristic = ethnicity_major
+  ) %>%
+  select(year, provider_name, characteristic_type, characteristic, measure, count)
+
+chars_parquet_ethnicity <- chars_parquet %>%
+  filter(ethnicity_major != "Total") %>%
+  mutate(
+    characteristic_type = "Ethnicity",
+    characteristic = ethnicity_major
+  ) %>%
+  select(year, provider_name, characteristic_type, characteristic, measure, count)
+
+
+chars_parquet <-
+  rbind(
+    chars_parquet_total_age, chars_parquet_age,
+    chars_parquet_total_sex, chars_parquet_sex,
+    chars_parquet_total_lldd, chars_parquet_lldd,
+    chars_parquet_total_ethnicity, chars_parquet_ethnicity
+  )
```

Comment:
I think this pivoting code might be better done in the R/data-prep/create_data_files.R script.

That way you can run it once now, save a new version of the parquet data file, and then whenever someone uses the app they load that file straight in, rather than all of these lines running every time the app is used?

You'll need to run the SQL query for this data separately and then save a CSV version in the data folder so line 21 with `apps_demographics <- ...` will read it in, and then do this code on that apps_demographics table, before then changed line 42 where it writes out the parquet data set to write out the new table you've pivotted to (characteristics_no_total)